### PR TITLE
[WIP] Add environment variables to kpack image CR

### PR DESCRIPTION
**Pull request by kramerul**
Thanks for contributing to cloud_controller_ng. To speed up the process of reviewing your pull request please provide us with:

* A short explanation of the proposed change:

See 

* An explanation of the use cases your change solves

[Env variables are not passed to kpack builds](https://github.com/cloudfoundry/capi-k8s-release/issues/37)

* Links to any other associated PRs

* [X] I have reviewed the [contributing guide](https://github.com/cloudfoundry/cloud_controller_ng/blob/master/CONTRIBUTING.md)

* [X] I have viewed, signed, and submitted the Contributor License Agreement

* [X] I have made this pull request to the `master` branch

* [X] I have run all the unit tests using `bundle exec rake`

* [ ] I have run [CF Acceptance Tests](https://github.com/cloudfoundry/cloud_controller_ng#cf-acceptance-tests-cats)

### CODE REVIEW

File: lib/cloud_controller/kpack/stager.rb [hunk 1]
```diff
@@ -75,6 +75,11 @@ def image_resource(staging_details)
             blob: {
               url: blobstore_url_generator.package_download_url(staging_details.package),
             }
+          },
+          build: { 
+            env: staging_details.environment_variables.to_a.
+                  select{ | key, value | key =~ /^BP[A-Z]{0,1}_/ }.
```
Comment (aashah) [inline lib/cloud_controller/kpack/stager.rb:81]: I'm curious why there is a whitelist on environment variables prefixed with `BP*`?

**Comment by aashah** [comment]
Hey @kramerul,

Thanks for submitting this PR. There's actually a prioritized story ([link](https://www.pivotaltracker.com/story/show/172301179)) near the top of our backlog to do this! I'm sure whoever on the team picks it up will be happy to know there's a PR  :)

I see you've marked it as WIP as well, what other work is on your TODO list?

- Aakash

File: lib/cloud_controller/kpack/stager.rb (hunk 1 above)
Comment (kramerul) [reply lib/cloud_controller/kpack/stager.rb:81]: This white list is not really required. It was only a gut feeling that feeding in variables like `VCAP_SERVICES` could cause problems. We also had these kind of problems when the `to_s` was missing in the line below.

**Comment by kramerul** [comment]
Hi @aashah ,

I marked this PR as WIP, because I never filed a PR for cloud controller and I wasn't sure if I needed to run the acceptance tests for this rather small PR. 
For the acceptance tests I would have needed a running CloudFoundry, which is quite a lot of work for me, since we are only dealing with `cf-for-k8s` at the moment. I have no experience with BOSH at all.

File: lib/cloud_controller/kpack/stager.rb (hunk 1 above)
Comment (aashah) [reply lib/cloud_controller/kpack/stager.rb:81]: Yeah `VCAP_SERVICES` is not going to be the future method that bound services supply credentials to the app. There is upcoming work for this here - https://www.pivotaltracker.com/story/show/172562308

I think I'd still suggest relaxing the constraint, and maybe only filter out `VCAP_SERVICES`. There are many other environment variables that app developers set, and `BP*` is overly restrictive (and perhaps too specific to a single buildpack like bellsoft-liberica buildpack).

**Comment by aashah** [comment]
Ah yeah, that's totally fine! Definitely agree, CATs is not required for this PR. Once we resolve some of the discussion on that "constraint" in the code, I think I can check out the PR and try it out!

**Comment by aashah** [comment]
It also looks like there are several rubocop errors detailed here - https://travis-ci.org/github/cloudfoundry/cloud_controller_ng/jobs/680412878

**Comment by kramerul** [comment]
Hi @aashah,

I removed the select for the variables starting with BP*.

**Comment by aashah** [comment]
Hey @kramerul,

I think your original gut feeling around wanting to prevent `VCAP_SERVICES` from being passed in resonates with me. Echo'ing some of the feedback from above, could you keep & update the filter to only prevent `VCAP_SERVICES`?

Additionally, there are many silly rubocop errors that need to be taken care of before I can merge the PR - https://travis-ci.org/github/cloudfoundry/cloud_controller_ng/jobs/681337312

- Aakash

**Comment by kramerul** [comment]
Hi @aashah,

I'm now filtering out `VCAP_SERVICES` and also fixed the rubocop issues.

**Comment by aashah** [comment]
Looks good to me!

Dev acceptance:

```
$ cf push --no-start node
Pushing app node to org o / space s as admin...
Getting app info...
Creating app with these attributes...
...
memory usage:   1024M
     state   since                  cpu    memory   disk     details
#0   down    2020-05-08T23:08:33Z   0.0%   0 of 0   0 of 0
```

```
$ cf set-env node FOO BAR
Setting env variable 'FOO' for app node in org o / space s as admin...
OK
TIP: Use 'cf restage node' to ensure your env variable changes take effect

$ cf start node
Starting app node in org o / space s as admin...

Staging app and tracing logs...

...
   Build successful
...

type:           web
instances:      1/1
memory usage:   1024M
     state     since                  cpu    memory    disk      details
#0   running   2020-05-08T23:09:48Z   0.0%   0 of 1G   0 of 1G
```

```
$ k get build eeb7efc2-9c97-48f4-99e2-dd008ca4917f-build-1-2dr9c -n cf-workloads-staging -o yaml
apiVersion: build.pivotal.io/v1alpha1
kind: Build
metadata:
...
spec:
  env:
  - name: FOO
    value: BAR
...
```

**Pull request merged**